src/domain/ledger/activity-checker.ts

```diff
@@ -0,0 +1,38 @@
+import { toSats } from "@domain/bitcoin"
+
+const MS_PER_HOUR = (60 * 60 * 1000) as MilliSeconds
+const MS_PER_DAY = (24 * MS_PER_HOUR) as MilliSeconds
+const MS_PER_30_DAYS = (30 * MS_PER_DAY) as MilliSeconds
+
+export const ActivityChecker = ({
```

Comment:
I would either:
- rename this to `MonthlyActivityChecker`, or

- pass the `monthlyVolumeThreshold` argument into a sub-function called `above30DayThreshold`/`aboveMonthlyThreshold` here instead of into the top-level function
 
just to make this concept a bit clearer.